internal/services/mssql/mssql_server_dns_alias_resource_test.go

```diff
@@ -0,0 +1,138 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type ServerDNSAliasResource struct{}
+
+func TestAccServerDNSAlias_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_mssql_server_dns_alias", "test")
+	r := ServerDNSAliasResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccServerDNSAlias_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_mssql_server_dns_alias", "test")
+	r := ServerDNSAliasResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (r ServerDNSAliasResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := parse.ServerDNSAliasID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.MSSQL.ServerDNSAliasClient.Get(ctx, id.ResourceGroup, id.ServerName, id.DnsAliaseName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retreiving %s: %v", id, err)
+	}
+	if utils.ResponseWasNotFound(resp.Response) {
+		return utils.Bool(false), nil
+	}
+	return utils.Bool(true), nil
+}
+
+// Configs
+
+var sf = fmt.Sprintf
+
+func (r ServerDNSAliasResource) vars(data *acceptance.TestData) string {
+	return sf(`
+provider "azurerm" {
+  features {}
+}
+
+variable "rg" {
+  type        = string
+  description = "resource group to use"
+  default     = "acctestRG-appServerDNSAlias-%[1]d"
+}
+
+variable "loc" {
+  type        = string
+  description = "location of resource"
+  default     = "%[2]s"
+}
+
+variable "sql_dns_alias" {
+	type = string
+	default = "acctest-dns-alias-%[1]d"
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = var.rg
+  location = var.loc
+}
+
+resource "azurerm_mssql_server" "sql" {
+  administrator_login          = "umtacc"
+  administrator_login_password = "random81jdpwd_$#fs"
+  location                     = var.loc
+  name                         = "acctestrg-sql-sever-%[1]d"
+  resource_group_name          = azurerm_resource_group.test.name
+  version                      = "12.0"
+}
+`, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r ServerDNSAliasResource) basic(data acceptance.TestData) string {
+	s := fmt.Sprintf(`
+%s
+
+resource "azurerm_mssql_server_dns_alias" "test" {
+  server_name         = azurerm_mssql_server.sql.name
+  resource_group_name = azurerm_resource_group.test.name
+	dns_alias_name 			= var.sql_dns_alias
```

Comment:
The config seems not correctly formated, please use https://github.com/katbyte/terrafmt to format this file.